provider/bidengine/order.go

```diff
@@ -404,7 +404,13 @@ func (o *order) shouldBid(group *dtypes.Group) (bool, error) {
 
 	// does provider have required attributes?
 	if !group.GroupSpec.MatchAttributes(o.session.Provider().Attributes) {
-		o.log.Debug("unable to fulfill: incompatible attributes")
+		o.log.Debug("unable to fulfill: incompatible provider attributes")
+		return false, nil
+	}
+
+	// does order have required attributes?
+	if !o.cfg.Attributes.SubsetOf(group.GroupSpec.Requirements.Attributes) {
```

Comment:
What's the difference between the session and the cfg ?

Reply:
`session` gives the attributes stored on-chain when the provider was created with `$akash tx provider create`. If an order specifies some required attributes, then the session's provider attributes must have those attributes, otherwise it can't bid on the order.

On the other hand, `cfg` gives the attributes that the order must have for the provider to bid on it. These are specified using `$akash provider run --provider-config <path-to-conf-file>`.

Reply:
I don't actually see this `--provider-config` flag, was this added recently?

Reply:
Yes. It's added in this PR.